src/WebJobs.Extensions.DurableTask/SystemEnvironment.cs

```diff
@@ -0,0 +1,80 @@
+    {
+        // Enviroment variable names. Consider moving these to a separate file if they get too big.
+        // Duplicated from: https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Environment/EnvironmentSettingNames.cs
+        // `FunctionsLogsMountPath` is set dynamically by the platform when it is safe to specialize the host instance (eg. file system is ready)
+        public const string ContainerName = "CONTAINER_NAME";
+        public const string AzureWebsiteInstanceId = "WEBSITE_INSTANCE_ID";
+        public const string FunctionsLogsMountPath = "FUNCTIONS_LOGS_MOUNT_PATH";
+
+        /// <summary>
+        /// Private and lazily constructed singleton instance of SystemEnvironment.
+        /// </summary>
+        private static readonly Lazy<SystemEnvironment> PrivInstance =
+            new Lazy<SystemEnvironment>(() => new SystemEnvironment());
+
+        /// <summary>
+        /// Private constructor, clients should obtain an instance via the `Instance` static attribute.
+        /// </summary>
+        private SystemEnvironment()
+        {
+        }
+
+        /// <summary>
+        /// The singleton instance of SystemEnvironment, no explicit constructor is exported.
+        /// </summary>
+        public static SystemEnvironment Instance => PrivInstance.Value;
+
+        /// <summary>
+        /// Gets a value indicating whether the application is running in a Linux Consumption (dynamic)
+        /// App Service environment.
+        /// </summary>
+        /// <returns>true if running in a Linux Consumption App Service app; otherwise, false.</returns>
+        public bool IsLinuxConsumtpion()
+        {
+            return !this.IsAppService() && !string.IsNullOrEmpty(this.GetEnvironmentVariable(ContainerName));
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether the application is running in a Linux App Service
+        /// environment (Dedicated Linux).
+        /// </summary>
+        /// <returns>true if running in a Linux Azure App Service; otherwise, false.</returns>
+        public bool IsLinuxDedicated()
+        {
+            return this.IsAppService() && !string.IsNullOrEmpty(this.GetEnvironmentVariable(FunctionsLogsMountPath));
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether the application is running in App Service
+        /// (Windows Consumption, Windows Dedicated or Linux Dedicated).
+        /// </summary>
+        /// <returns>true if running in a Azure App Service; otherwise, false.</returns>
+        private bool IsAppService()
+        {
+            return !string.IsNullOrEmpty(this.GetEnvironmentVariable(AzureWebsiteInstanceId));
+        }
+
+        /// <summary>
+        /// Retrieves the value from an environment variable from the current process.
+        /// </summary>
+        public string GetEnvironmentVariable(string name)
+        {
+            return Environment.GetEnvironmentVariable(name);
```

Comment:
This should use `INameResolver` to make it easier to test.

Reply:
@ConnorMcMahon , you mentioned in a later comment (which for some reason I can't comment on) that you recommended that I refactor this to use `INameResolver`. I've been very unlucky at finding information about that pattern, for instance [this](https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.webjobs.inameresolver?view=azure-dotnet) page has no such description. 

Two questions:
(1) What should I know about it?
(2) As a .NET/C# newcomer, where can I find this info? I swear I've been googling for it and results haven't been very helpful in giving me a straightforward answer. Oh well, maybe this particular query was unlucky.....

Reply:
Nevermind these questions, I asked in a C# chat and ended up getting answers to these two questions :)